train/tictactoe.py

```diff
@@ -46,14 +46,26 @@ def _get_input_shape(self, cfg):
         return shape_cpu if cfg["cpu"] else shape_gpu
 
     def _create_model_simple_two_headed(self, cfg):
+        l2reg = cfg["model"]["l2reg"]
+
         inputs = Input(
-            shape=self._get_input_shape(cfg),
+             shape=self._get_input_shape(cfg),
             name="input_planes")
         flow = tf.keras.layers.Flatten()(inputs)
         flow = Dense(units=9, activation="relu")(flow)
         head_val = Dense(
             units=1, activation="tanh", name="value_head")(flow)
         head_probs = Dense(units=self.MOVE_NUM, name="policy_head")(flow)
+        flow = Dense(units=27, activation="relu", kernel_regularizer=l2reg)(flow)
+        flow = Dense(units=27, activation="relu", kernel_regularizer=l2reg)(flow)
+        flow = Dense(units=27, activation="relu", kernel_regularizer=l2reg)(flow)
+        head_val = Dense(units=27, activation="relu", kernel_regularizer=l2reg)(flow)
+
+        head_val = Dense(units=27, activation="relu", kernel_regularizer=l2reg)(head_val)
+        head_val = Dense(units=1, activation="tanh", name="value_head", kernel_regularizer=l2reg)(head_val)
+        head_probs = Dense(units=27, activation="relu", kernel_regularizer=l2reg)(flow)
+        head_probs = Dense(units=27, activation="relu", kernel_regularizer=l2reg)(head_probs)
+        head_probs = Dense(units=self.MOVE_NUM, name="policy_head", kernel_regularizer=l2reg)(head_probs)
```

Comment:
this is a bit of mess, lets use 'flow' for the shared layers, 'flow_val' and 'flow_probs' for the two heads flows, and 'head_val' and 'head_probs' only once